File: binary_search_repetition.py
from typing import Tuple

def binary_search(arr: list[int], item: int) -> Tuple[bool, int]: # add list[int]
  low = 0
  high = len(arr) - 1

  if not arr:
    return False, -1

  while low <= high:
    midd = (low + high) // 2
    guess = arr[midd]

    if guess == item:
      return True, midd
    elif guess > item:
      high = midd - 1
    else:
      low = midd + 1
  
  return False, -1


print(binary_search([2, 4, 6, 7, 8, 69], 8))

File: binary_search_repetition.js
'use strict';

const binarySearch = (arr, item) => {
  let low = 0;
  let high = arr.length - 1;

  if (arr.length === 0) return -1; // проверка вне цикла

  while (low <= high) {

    let mid = Math.floor((low + high) / 2);
    let guess = arr[mid];

    if (guess === item) return mid;

    if (guess > item) high = mid - 1;

    if (guess < item) low = mid + 1;
  }
  return -1;
}

console.log(binarySearch([2, 4, 6, 7, 8, 69], 8))
console.log(binarySearch([2, 4, 6, 7, 8, 69], 90))